GTFS_converter.py

```diff
@@ -0,0 +1,241 @@
+                travel_times[pair].append(travel_time)
+    
+        prev_stop = {
+            'trip_id': trip_id,
+            'stop_id': stop_id,
+            'departure_time': row['departure_time'],
+            'sequence': sequence
+        }
+    average_travel_times = [[pair[0], pair[1], sum(times)/len(times)] for pair, times in travel_times.items()]
+    file.close()
+    os.remove('modified_stop_times.txt')
+
+#here the dictionaries of stations and edges are created 
+routes_list=[]
+processed_list_of_stations=[]
+compiled_list_of_routes=[]
+for list_of_stations in ordered_list_of_stations: #here we create the database of stations
+    if list_of_stations in processed_list_of_stations: #if the path has already been processed
+        routes_list=[]
+        continue
+    elif list_of_stations[::-1] in processed_list_of_stations: #if the reverse of this path has been processed, a second track is added 
+        for index, station in enumerate(list_of_stations):
+            stations['Station'+str(counter)]={'type':'SimpleStation','name':str(station)+"_2",'u':str(station)+"_1",'v':str(station)+"_2",'key':2,'args':{'mean_boarding': 20,'mean_alighting': 20}}
+            routes_list.append(['u:'+str(station)+"_2",'v:'+str(station)+"_1",'key:'+"2"])
+            if index==0:
+                pass
+            else:
+                previous_station=list_of_stations[index-1]
+                for travel_time in average_travel_times:
+                    if travel_time[0]==previous_station and travel_time[1]==station:
+                        average_travel_time=travel_time[2]
+                        break
+                edges['Edge'+str(counter)]={'type':'TimedTrack','u':str(previous_station)+"_2",'v':str(station)+"_1",'key':2,"traversal_time": average_travel_time}
+                routes_list.insert(-1,['u:'+str(previous_station)+"_1",'v:'+str(station)+"_2",'key:'+"2"])
+            counter=counter+1
+        compiled_list_of_routes.append(routes_list)
+        routes_list=[]
+        continue    
+    else:
+        for index, station in enumerate(list_of_stations): #base case, the path hasnt been processed yet
+            stations['Station'+str(counter)]={'type':'SimpleStation','name':str(station)+"_1",'u':str(station)+"_1",'v':str(station)+"_2",'key':1,'args':{'mean_boarding': 20,'mean_alighting': 20}}
+            routes_list.append(['u:'+str(station)+"_1",'v:'+str(station)+"_2",'key:'+"1"])
+            if index==0:
+                pass
+            else:
+                previous_station=list_of_stations[index-1]
+                for travel_time in average_travel_times:
+                    if travel_time[0]==previous_station and travel_time[1]==station:
+                        average_travel_time=travel_time[2]
+                        break
+                edges['Edge'+str(counter)]={'type':'TimedTrack','u':str(previous_station)+"_2",'v':str(station)+"_1",'key':1,"traversal_time": average_travel_time}

```

Comment:
Traversal time should be an integer. Also cross-reference other component parameters where needed.

https://spur-sim.readthedocs.io/en/latest/reference/core.html#spur.core.component.TimedTrack